proto/abis/massa/abi/v1/abi.proto

```diff
@@ -13,65 +18,54 @@ option php_namespace = "Com\\Massa\\Abi\\V1";
 option ruby_package = "Com::Massa::Abi::V1";
 option swift_prefix = "MABI";
 
-// Address
-message Address {
-  // Address is a string representation of the address
-  string address = 1;
-}
-
-// Amount
-message Amount {
-  // Amount is a 64-bit unsigned integer
-  fixed64 amount = 1;
+// Error message used in abi Response
+message Error {
+  // a string representing the error
+  string message = 1;
 }
 
-// Empty
-message Empty {}
-
-// CreateSC
 message CreateSCRequest {
   // Bytecode is the compiled code of the smart contract
   bytes bytecode = 1;
 }
 
-// CreateSCResponse
 message CreateSCResponse {
-  // Address is a string representation of the address
-  Address address = 1;
+  // Address of the just created smart contract
+  massa.model.v1.NativeAddress sc_address = 1;
 }
 
 // CallSC
 message CallRequest {
-  // Address is the address of the smart contract
-  Address address = 1;
-  // Function is the name of the function to call
-  string function = 2;
-  // Arg is the argument to the function
-  bytes arg = 3;
+  // Address of the smart contract to call
+  massa.model.v1.NativeAddress target_sc_address = 1;
+  // Function to call in the targeted smart contract
+  string target_function_name = 2;
+  // Argument to the function serialized in a byte array.
+  bytes function_arg = 3;
   // call_coins is the amount of coins to pay for the call
-  Amount call_coins = 4;
+  massa.model.v1.NativeAmount call_coins = 4;
 }
 
 // CallResponse
 message CallResponse {
   // Return_data is the return value of the function
-  bytes return_data = 1;
+  bytes data = 1;
 }
 
 // LocalCall
 message LocalCallRequest {
-  // Address is the address of the smart contract
-  Address address = 1;
-  // Function is the name of the function to call
-  string function = 2;
-  // Arg is the argument to the function
-  bytes arg = 3;
+  // Address of the smart contract to call
+  massa.model.v1.NativeAddress target_sc_address = 1;
+  // Function to call in the targeted smart contract
+  string target_function_name = 2;
+  // Argument to the function serialized in a byte array.
+  bytes function_arg = 3;
 }
 
 // LocalCallResponse
 message LocalCallResponse {
   // Return_data is the return value of the function
-  bytes return_data = 1;
+  bytes data = 1;
 }
 
 // GenerateEventRequest
```

Comment:
use bytes in GenerateEvent but have the default SDK take strings and convert them to UTF8 when calling the ABI

Reply:
@damip 
just for information, the current wip implementation of this ABI in the AS SDK already does the conversion from UTF16 (user side) <--> UTF8 (abi side)
basically it's just adding a `to_vec`

Reply:
Basically in the SC SDK we would take a utf16 string as argument and convert it to utf8 then to bytes before sending it to the abi

Reply:
That's what I was trying to say :)